src/pages/Detail/Detail.js

```diff
@@ -0,0 +1,181 @@
+import axios from 'axios'
+import React, { useCallback, useEffect, useState } from 'react'
+import { Link, useParams } from 'react-router-dom'
+import AbilityOfPokemon from './components/AbilityOfPokemon'
+import VarietyOfPokemon from './components/VarietyOfPokemon'
+import './styles.css'
+
+export default function Detail() {
+  const params = useParams()
+
+  const id = params.id || 1
+
+  // save state detail pokemon.
+  const [detailPokemon, setDetailPokemon] = useState(() => ({
+    id: '',
+    name: '',
+    frontDefault: '',
+    types: '',
+    baseExperience: '',
+    weight: '',
+    height: '',
+    abilities: [],
+    stats: [],
+    speciesUrl: ''
+  }))
+
+  const [evolutionChain, setEvolutionChain] = useState(() => ({}))
+
+  // get detail of pokemon.
+  const fetchDetailPokemon = useCallback(url => {
+    return axios.get(url).then(response => response.data)
+  }, [])
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(async () => {
+    const species = await axios.get(`${detailPokemon.speciesUrl}`)
+    const resEvolutionChain = await axios.get(
+      `${species.data.evolution_chain?.url}`
+    )
+    const data = resEvolutionChain.data?.chain
+    const speciesOne = data.species.name
+    const urlOfSpeciesOne = data.species.url
+
+    const speciesTwo = data.evolves_to[0]?.species.name
+    const urlOfSpeciesTwo = data.evolves_to[0]?.species.url
+
+    const speciesThree = data.evolves_to[0]?.evolves_to[0]?.species.name
+    const urlOfSpeciesThree = data.evolves_to[0]?.evolves_to[0]?.species.url
```

Comment:
we should create a data parser for api response. the responded data can be in snake_case BUT data in our project should be in camelCase

Reply:
your mean of 'create a data parser' are make a skeleton data, right?
example:
`
const species = {
   name: "",
   url: "",
}
`